VideoLocker/src/main/java/org/edx/mobile/player/PlayerFragment.java

```diff
@@ -975,15 +978,21 @@ private boolean isScreenLandscape() {
         @Override
         public void run() {
             if (isResumed() && !isRemoving()) {
-                hideProgress();
-                if(player!=null) {
+                if (player != null) {
                     player.unfreeze();
+                    hideProgress();
                     if (player.isPlaying()) {
-                        hideProgress();
                         updateController("player unfreezed");
                     }
 
-                    if (pauseDueToDialog){
+                    if (IS_AUTOPLAY_ENABLED && isPrepared && !isPaused) {
+                        player.start();
+                    } else {
+                        updateController("player paused");
+                    }
```

Comment:
there's a comment on one of the callbacks call which reads:
>// start playback after 300 milli seconds, so that it works on HTC One, Nexus5, S4, S5
 // some devices take little time to be ready

So this seems fine i guess.